src/one/nio/os/systemd/SystemdNotify.java

```diff
@@ -49,9 +49,7 @@ public class SystemdNotify {
     public static void notify(String state) throws IOException {
         String notifySocket = System.getenv(NOTIFY_SOCKET_ENV);
         if (notifySocket == null) {
-            if (log.isDebugEnabled()) {
-                log.debug(NOTIFY_SOCKET_ENV + " environment variable is not defined");
-            }
+            log.debug(NOTIFY_SOCKET_ENV + " environment variable is not defined");
```

Comment:
Let's use `{}` placeholder to eliminate possible `String` allocation when debug is disabled.

Reply:
I did not use it here since the expression is a static string (`NOTIFY_SOCKET_ENV` is a compiletime constant thus the result of concatenation is too).
![image](https://user-images.githubusercontent.com/7693005/236284479-cbe5d2cd-aad7-42d9-bd8f-c9575ef95ab2.png)

Reply:
I get your point, it just doesn't look idiomatic.